src/sql/workbench/services/tableDesigner/browser/tableDesignerMetadata.ts

```diff
@@ -0,0 +1,35 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export class TableDesignerMetadata {
+
+	// Allowed metadata for every provider
+	public static mssqlAllowedMetdata: Set<string> = new Set(['isNode', 'isEdge', 'isSystemVersioned']);
+
+	// Provider ID to allowed metadata list set mapping
+	public static providerMetadataMap: Map<string, Set<string>> = new Map<string, Set<string>>([
+		['MSSQL', TableDesignerMetadata.mssqlAllowedMetdata]
+	]);
+
+	/**
+	 *
+	 * @param providerId provider ID for the table designer provider
+	 * @param metadata incoming metadata from the table designer provider
+	 * @returns filtered metadata with only allowed metadata
+	 */
+	public static getValidMetadata(providerId: string, metadata: { [key: string]: string }): { [key: string]: string } {
```

Comment:
getTelemetryInfo

Reply:
why telemetry info? this is the metadata as we discussed everywhere

Reply:
it is metadata until now, but this file is specifically only for telemetry. the allow list means properties allowed to be included in telemetry.